### PR TITLE
Handle no matching tasks

### LINKED ISSUE #9: Error when filtering returns no results
If you specify a filter, like this:

```
tasklist({
  filter: ["Imagename eq httpd.exe"]
})
```

When no matches are found, the promise is rejected with an error:

```
TypeError: Cannot read property 'replace' of undefined
    at data.map.task (D:\wamp\node_modules\tasklist\index.js:53:40)
    at Array.map (native)
    at pify.then.then.data (D:\wamp\node_modules\tasklist\index.js:
    at process._tickCallback (internal/process/next_tick.js:109:7)
```

Expected result is the promise being resolved with an empty array.

### PR DESCRIPTION
**Pull request by MarkTiedemann**
Fixes #9.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -45,17 +45,33 @@ module.exports = opts => {
...
-				task.cpuTime = sec(task.cpuTime);
-			}
-
-			return task;
-		}));
+		// INFO: No tasks are running which match the specified criteria.
+		// See: https://github.com/sindresorhus/tasklist/issues/9
+		.then(stdout => stdout.startsWith('INFO:') ?
+			[] :
+			neatCsv(stdout, {headers})
```
Comment (kevva) [inline index.js]: This doesn't improve readability that much, keep it on one line instead:

```js
.then(stdout => stdout.startsWith('INFO:') ? [] : neatCsv(stdout, {headers}))
```

File: index.js [hunk 2]
```diff
@@ -45,17 +45,33 @@ module.exports = opts => {
...
+						}
+
+						return task;
+					} catch (err) {
+						// Handle parsing errors gracefully
+						reject(err);
+					}
+				})
+			)
+		)
```
Comment (kevva) [inline index.js]: Revert all this. In case the `Array` is empty (which it should be if we encounter the issue described in #9) it won't do any parsing anyway.

File: test.js [hunk 3]
```diff
@@ -42,3 +42,10 @@ const macro = async (t, options) => {
 test('default', macro, {});
 test('verbose option', macro, {verbose: true});
 test('filter option', macro, {filter: ['status eq running', 'username ne F4k3U53RN4M3']});
+
+test('test handle no matching tasks gracefully', async t => {
+	const tasks = await tasklist({
+		filter: ['imagename eq does-not-exist']
+	});
+	t.true(tasks.length === 0);
```
Comment (kevva) [inline test.js]: `t.is(tasks.length, 0)`

**Review by kevva — CHANGES_REQUESTED** [review]

File: index.js (hunk 1 above)
Comment (MarkTiedemann) [reply index.js]: Including indentation, that line would be longer than 80 characters, which is too long IMHO. That's why I added the line breaks.

File: index.js (hunk 2 above)
Comment (MarkTiedemann) [reply index.js]: This part is not strictly related to #9. It is a safety measure in case any of the input returned by `tasklist` is not as we expect it to be, for whatever reason. E.g. it could be that `tasklist` does not output the `INFO:` part on Chinese system, or perhaps it returns a differently formatted `cpuTime` on an Arabic system, or perhaps the `memUsage` is not a number but some string in older Windows versions, or whatever. My point is that we do parsing there, and since neither of us know the source code of `tasklist`, we cannot be confident that it doesn't break in weird edge cases. That's why I prefer to have this extra try-catch that rejects the Promise on errors, since a Promise should never, in any case, throw but always reject.

File: test.js (hunk 3 above)
Comment (MarkTiedemann) [reply test.js]: Good catch! Will change that ASAP.

File: index.js (hunk 2 above)
Comment (kevva) [reply index.js]: Throwing inside a `Promise` will cause it to reject so this part is moot anyway.

Comment (MarkTiedemann) [reply index.js]: You're right, totally forgot about that!

EDIT: Removed silly comment I made (while coding at 2 am at the morning).

Comment (kevva) [reply index.js]: It's inside the `Promise` anyway since you're in a `.then()` :). Try running this:

```js
Promise.resolve()
	.then(() => {
		throw new Error('foo');
	})
	.catch(err => {
		console.log(err.message);
		//=> 'foo'
	});
```

**Comment by MarkTiedemann** [comment]
@kevva I made some stylistic changes, as you requested.

Hope this PR can finally be accepted now.

**Comment by sindresorhus** [comment]
Thank for fixing @MarkTiedemann :)
